src/app/(shop)/products/sitemap.ts

```diff
@@ -0,0 +1,21 @@
+import { siteUrl } from '@/config';
+import { getAllProductPageSummaryNodes } from '@/features/ProductPage/data';
+import * as Sentry from '@sentry/nextjs';
+import { MetadataRoute } from 'next';
+
+export default async function sitemap(): Promise<MetadataRoute.Sitemap> {
+  try {
+    // Google's limit is 50,000 URLs per sitemap
+    const summaryNodes = await getAllProductPageSummaryNodes();
```

Comment:
Note that these clients were consolidated, so we could make the same queries in the page component and here. Next will de-dupe those queries so we're only making this expensive query once.